### PR TITLE
14 configure a users s3 bucket as the destination for file transfer

### LINKED ISSUE #14: Configure a user's S3 bucket as the destination for file transfer
Given the user's access keys, users should be able to view the existing buckets within S3 and select a bucket as the destination for file transfer.

### PR DESCRIPTION
**Pull request by KevinMi2023p**
This splits the original window into 2 windows.

The login window looks like this now.
image

If a successful login happen it goes to this window, they'll still have the option to log out from here.
image

If a bucket exists that the user has access to then it'll go to this window.
image

Right now I just made a simple temporary window, but to reset everything it's the same thing as logging out. So when doing some test the extension doesn't get stuck in one of the windows.

PR resolves #14

### CODE REVIEW

File: src/aws_s3_bucket_front.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+        // User's keys for AWS S3
+        const bucketName = target.bucketName.value;
+
+        // Reuse the S3 object from App
+        const s3Auth = this.props.existingS3Obj!;
+
+        // Configure the bucket
+        s3Auth.changeBucket(bucketName);
+        s3Auth.checkAndDisplayValidBucket(this.props.onViewChange!, 'temp-window', this.props.onS3ObjChange!, s3Auth);
```
Comment (kpattaswamy) [inline src/aws_s3_bucket_front.tsx:44]: i'm currently working on adding more unit tests for changing bucket name, but some sort of alerting the user that an invalid bucket name was given should happen. maybe in another pr

also thoughts on a different function name? i think something like checkBucketAndChangeUI

File: src/aws_s3_bucket_front.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        const target = event.currentTarget.elements;
+
+        // User's keys for AWS S3
+        const bucketName = target.bucketName.value;
+
+        // Reuse the S3 object from App
+        const s3Auth = this.props.existingS3Obj!;
+
+        // Configure the bucket
+        s3Auth.changeBucket(bucketName);
```
Comment (kpattaswamy) [inline src/aws_s3_bucket_front.tsx:43]: i know `checkAndDisplayValidBucket` checks to see if the bucket is valid, but what are your thoughts on making `changeBucket` asserting the bucket string is non empty

File: src/aws_s3_bucket_front.tsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
+import React, {FormEvent} from 'react';
```
Comment (kpattaswamy) [inline src/aws_s3_bucket_front.tsx:1]: thoughts on a different file name? like aws_s3_config_bucket and different class name

**Review by kpattaswamy — COMMENTED** [review]
Looks good overall, some comments

File: src/aws_s3_bucket_front.tsx (hunk 1 above)
Comment (KevinMi2023p) [reply src/aws_s3_bucket_front.tsx:44]: I guess in a future pr we can outlight the extension red or something simple at first.

I can also change the name of the function.

File: src/aws_s3_bucket_front.tsx (hunk 2 above)
Comment (KevinMi2023p) [reply src/aws_s3_bucket_front.tsx:43]: I think that's something I'll try in a future PR

**Comment by KevinMi2023p** [comment]
I made changes based on the comments above

**Review by kpattaswamy — APPROVED** [review]
Looks good, later pr can do error handling

**Pull request merged**